### DISCUSSION TITLE
[Proposal] Remove experiment tracking from Kedro-Viz

### DISCUSSION THREAD

**Discussion by tynandebold** [kedro-org/kedro-viz#1831] [Ideas]
## Description

Experiment tracking should be removed from Kedro-Viz.

## Context

For a combination of reasons (competition from other products, an initial MVP release that wasn't viable, ineffective marketing, etc.), experiment tracking wasn't well adopted and usage has plateaued.

There's not much wrong with maintaining a feature that has lukewarm adoption and low complexity. Experiment tracking is a much higher-complexity feature that adds to and creates tech debt of its own while simultaneously introducing an entire new type of API.

Lastly, with the release of the Shareable URL feature, we're now able to host the Kedro-Viz flowchart page statically. We can't do this with experiment tracking, and have to filter out this feature when the app is hosted in this manner. Removing it would free up this maintenance and continue to open to door to a plethora of static hosting solutions.

#### Unique users, total

<img width="1414" alt="image" src="https://github.com/kedro-org/kedro-viz/assets/16869061/ba078d0f-e04e-4d3f-94c0-0064a88ae534">

Note here that in July 2022, we had a then-high number of 89 users, which hasn't been meaningfully surpassed since, nearly 1.5 years later. We also haven't seen any increase in adoption since we released [collaborative experiment tracking](https://github.com/kedro-org/kedro-viz/releases/tag/v6.2.0) on 26 May 2023.

---
#### Unique users, run selection

<img width="1404" alt="image" src="https://github.com/kedro-org/kedro-viz/assets/16869061/511f1a8b-5248-4c63-8c17-41059728ff4c">

The high of 20 in June 2022 hasn't been surpassed.

---

#### Unique users, toggle compare runs

<img width="1402" alt="image" src="https://github.com/kedro-org/kedro-viz/assets/16869061/bd948c60-92a6-4f0b-8531-4a89f09b42f0">

The high of 14 in March 2022 hasn't been surpassed.

---

#### Unique users, selection of `Metrics` tab (chart data dates back one year only)

<img width="1419" alt="image" src="https://github.com/kedro-org/kedro-viz/assets/16869061/744646b0-12ee-4573-b0f0-c79c505b36ec">

---

#### Unique users, download run data (chart data dates back one year only)

<img width="1436" alt="image" src="https://github.com/kedro-org/kedro-viz/assets/16869061/44256484-b5fb-4eb4-9ba5-f462dc83eada">

Very little engagement here.

---

## Possible Implementation

Remove all features related to experiment tracking and all code that powers those features.

## Possible Alternatives

Remove experiment tracking from Kedro-Viz and create a separate Kedro plugin with the same feature set. 

## Checklist

- [x] Include labels so that we can categorise your feature request

**Comment by lemonhead94** [comment]
I think the comments are valid especially for allowing static pages. However, I would like to see the incorporation of an remote MLflow endpoint. Than experiment tracking could simply pull at runtime from the mlflow server instance and the nice UI wouldn’t go to waste.

**Comment by sbrugman** [comment]
My user perspective: we use MLFlow as model registry, and prefer to use that for experiment tracking too, as models are outputs there. My personal perspective is that it's a scope creep in Kedro and could be completely removed.

**Comment by astrojuanlu** [comment]
We're removing experiment tracking. https://github.com/kedro-org/kedro-viz/issues/2202
